web/assets/data-loss-warning.test.js

```diff
@@ -0,0 +1,200 @@
+import { DataLossWarning } from './data-loss-warning'
```

Comment:
are we going to run these in GitHub? (I'd personally be fine with not having jest test and just covering the behaviour fully in cypress)

Reply:
I was aiming to have all this in unit tests but the quirkiness of jsdom has just made it not worthwhile. I'll add a couple of lines to cypress to test the dialog is actually visible then I think we can safely drop this.